lib/internal/repl/utils.js

```diff
@@ -87,7 +89,98 @@ function isRecoverableError(e, code) {
   }
 }
 
+// Appends the preview of the result
+// to the tty.
+function appendPreview(repl, result, cursorTo, clearScreenDown) {
+  repl.previewResult = `\u001b[90m // ${result}\u001b[39m`;
+  const line = `${repl._prompt}${repl.line} //${result}`;
+  const columns = repl.output.columns;
+  const hasColors = repl.output.hasColors ? repl.output.hasColors() : false;
+  const s = hasColors ?
+    `${repl._prompt}${repl.line}${repl.previewResult}` : line;
+
+  // Cursor to left edge.
+  cursorTo(repl.output, 0);
+  clearScreenDown(repl.output);
+
+  if (columns !== undefined) {
+    repl.output.write(line.length < columns ?
+      s : `${s.slice(0, columns - 3)
+        .replace(/\r?\n|\r/g, '')}...\u001b[39m`);
+  } else {
+    repl.output.write(s);
+  }
+
+  // Move back the cursor to the original position
+  cursorTo(repl.output, repl.cursor + repl._prompt.length);
+}
+
+function clearPreview(repl) {
+  if (repl.previewResult !== '') {
+    repl._refreshLine();
+    repl.previewResult = '';
+  }
+}
+
+// Called whenever a line changes
+// in repl and the eager eval will be
+// executed against the line using v8 session
+let readline;
+function makePreview(repl, eagerSession, eagerEvalContextId, line) {
+  const lazyReadline = () => {
+    if (!readline) readline = require('readline');
+    return readline;
+  };
+
+  const { cursorTo, clearScreenDown } = lazyReadline();
```

Comment:
:thinking: can we just `const { cursorTo, clearScreenDown } = require('readline')` at the top now?